storage/store/inmemory/unsynchronized/registers.go

```diff
@@ -72,3 +75,18 @@ func (r *Registers) Store(registers flow.RegisterEntries, height uint64) error {
 
 	return nil
 }
+
+func (r *Registers) AddToBatch(batch storage.ReaderBatchWriter) error {
+	for id, value := range r.store {
+		encodedHeight := make([]byte, 8)
+		binary.BigEndian.PutUint64(encodedHeight, r.blockHeight)
+		key := append(encodedHeight, id.Bytes()...)
```

Comment:
I think this should use the same method as the main implementation to generate the key